src/User/Infrastructure/MongoDBUserRepository.py

```diff
@@ -43,7 +43,17 @@ def get_by_id(self, _id: str) -> User:
 
     def update_user(self, updated_user: User) -> None:
         updated_user = updated_user.dict()
+        updated_user.pop("favorite_adoption_publications", None)
         updated_user["_id"] = ObjectId(updated_user["_id"])
+
+        user_documents = self.adoption_publications.find(
+            {"user._id": updated_user["_id"]}
+        )
```

Comment:
use update_many like this
result = collection.update_many({"user._id": updated_user["_id"]}, {"$set": {"user": updated_user}})